### DISCUSSION TITLE
Javascript alternative?

### DISCUSSION THREAD

**Discussion by hg0428** [JSBSim-Team/jsbsim#984] [Ideas]
I saw JS in the name and hoped that this was JS, but then I clicked on it and it wasn't.
Is there any way I could integrate this with THREE.js via WASM?

**Comment by seanmcleod** [comment]
Nope, they're part of the initials of the person who started the project :wink:.

I don't see any particular reason you couldn't compile JSBSim to WASM, although I'm not aware of anyone trying to do so.

**Reply by Felipegalind0** [reply]
[0x62](https://github.com/0x62) pulled it off, he made [jsbsim-wasm](https://github.com/0x62/jsbsim-wasm), and I used it to create a [flight sim that runs in the browser](https://github.com/0SFS/0SFS.github.io)

**Comment by hg0428** [comment]
It would be hard to get this to integrate with our current code, even via wasm. I guess I need a different solution. If anyone has ideas, please tell me.
It's looking like I will have to build my own.

**Reply by Felipegalind0** [reply]
nobody is going to 'give you ideas' unless you explain what you are building, maybe start there?

i managed to integrate this into [Babylon.js](https://github.com/BabylonJS/Babylon.js) and made a [open source flight sim](https://github.com/0SFS/0SFS.github.io) out of it

**Comment by 0x62** [comment]
I came across this thread looking for something similar. For anyone else that stumbles on it, I have published a WASM build: https://github.com/0x62/jsbsim-wasm

**Reply by Felipegalind0** [reply]
I used that to create a [flight sim](https://github.com/0SFS/0SFS.github.io) and while making it had to open a [PR](https://github.com/0x62/jsbsim-wasm/pull/8) in [0x62/jsbsim-wasm](https://github.com/0x62/jsbsim-wasm)
It seems @seanmcleod70 reached out to you to discuss up-streaming this, since your repo is MIT, I'm going to go ahead and do it on my JSBSim fork, PR into [JSBSim-Team/jsbsim ](https://github.com/JSBSim-Team/jsbsim) (credit u ofc)

**Comment by seanmcleod70** [comment]
@Felipegalind0 your flight sim running in the browser on top of wasm-JSBSim with the external view etc. looks great.

My personal interest in seeing a wasm-JSBSim package is to enable additional options for marimo style notebooks, given marimo's support for wasm-html exports.

I have created a number of marimo based notebooks related to flight dynamics, with a fair number of them making use of running JSBSim within the notebook in order to generate results. For users on the internet who don't want to download and have to install Python, JSBSim and marimo I do a static html snapshot export to - https://seanmcleod70.github.io/FlightDynamicsCalcs/

For those users interested in taking the notebook and modifying the parameters, modifying the code etc. the marimo source notebooks are also made available at - https://github.com/seanmcleod70/FlightDynamicsCalcs

So I'm looking for the 3rd option, for users who don't want to have to download and install things, but who do want to modify the notebook to see how the results change. So for this case I want to be able to host a wasm-html export of the notebooks on Github so that everything runs in the browser.

I have done this fairly successfully for some project work when the only dependencies are libraries like numpy, matplotlib etc. that are available as wasm packages via Pyodide.

For example I've exported one of the flight dynamics notebook as wasm-html since it isn't dependent on JSBSim, see - https://seanmcleod70.github.io/FlightDynamicsCalcs/CrossWindCrabAngle.html

So with wasm-jsbsim via Pyodide I could instead of exporting static html snapshots simply export wasm-html exports for all of the notebooks.
